overlord/devicestate/devicemgr.go

```diff
@@ -142,6 +142,7 @@ func Manager(s *state.State, hookManager *hookstate.HookManager, runner *state.T
 	runner.AddHandler("mark-preseeded", m.doMarkPreseeded, nil)
 	runner.AddHandler("mark-seeded", m.doMarkSeeded, nil)
 	runner.AddHandler("setup-run-system", m.doSetupRunSystem, nil)
+	runner.AddHandler("ensure-next-boot-to-run-mode", m.doEnsureNextBootToRunMode, nil)
```

Comment:
this name doesn't quite encapsulate that this is going to restart as well?   maybe restart-system-to-run-mode?